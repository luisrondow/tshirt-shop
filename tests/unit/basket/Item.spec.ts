import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { mockNuxtImport } from '@nuxt/test-utils/runtime'

import { createTestingPinia } from '@pinia/testing'

import { BasketItem } from '#components'

mockNuxtImport('useFetch', () => {
  return () => ({ data: {}, refresh: () => {} })
})

describe('BasketItem component', () => {
  it('should render correctly', () => {
    const wrapper = mount(BasketItem, {
      global: {
        plugins: [createTestingPinia()],
      },
      props: {
        id: 'id-1',
        name: 'Product 1',
        image: 'https://via.placeholder.com/150',
        quantity: 1,
        price: 100,
        stock: 10,
      },
    })

    expect(wrapper.text()).toContain('Product 1')
    expect(wrapper.text()).toContain('â‚¬100')
    expect(wrapper.find('img').attributes('src')).toBe('https://via.placeholder.com/150')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should call removeFromBasket store event', async () => {
    const wrapper = mount(BasketItem, {
      global: {
        plugins: [createTestingPinia({ stubActions: false })],
      },
      props: {
        id: 'id-1',
        name: 'Product 1',
        image: 'https://via.placeholder.com/150',
        quantity: 1,
        price: 100,
        stock: 10,
      },
    })

    const basketStore = useBasket()

    await wrapper.find('button').trigger('click')

    expect(basketStore.removeFromBasket).toHaveBeenCalled()
  })
})
