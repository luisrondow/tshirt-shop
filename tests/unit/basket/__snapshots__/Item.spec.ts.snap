// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BasketItem component > should render correctly 1`] = `
"<div class="flex w-full gap-4" quantity="1"><img class="h-48 w-32" src="https://via.placeholder.com/150" alt="Product 1 photo">
  <div class="flex w-full items-center justify-between">
    <div class="flex flex-col justify-between gap-24">
      <h2 class="font-serif text-3xl">Product 1</h2><span class="text-xl">â‚¬100.00</span>
    </div>
    <div class="flex flex-col items-end gap-6 md:flex-row">
      <div>
        <p class="mb-4">Quantity:</p>
        <div class="dropdown relative">
          <div data-testid="dropdown-select" class="flex h-10 min-w-24 cursor-pointer items-center rounded border border-gray-200 bg-white p-3"><span>1</span><svg xmlns="http://www.w3.org/2000/svg" class="ml-auto h-4 w-4" fill="none" viewBox="0 0 24 24" stroke="currentColor">
              <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M19 9l-7 7-7-7"></path>
            </svg></div>
          <!--v-if-->
        </div>
      </div><button class="h-10 w-full uppercase border border-black bg-white px-4 py-2 text-black hover:bg-black hover:text-white h-auto">Remove</button>
    </div>
  </div>
</div>"
`;
